driver/ppm_fillers.c

```diff
@@ -701,9 +701,7 @@ static int compat_accumulate_argv_or_env(compat_uptr_t argv,
 
 #endif
 
-// probe_kernel_read() only added in kernel 2.6.26
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 26)
-long probe_kernel_read(void *dst, const void *src, size_t size)
+long probe_kernel_read_old(void *dst, const void *src, size_t size)
```

Comment:
This can stay under the `#ifdef` guard, right? So tl;dr, we'd do:

```
if < 2.6.26
  long copy_from_kernel_nofault(...)
else if < 5.8.0
  #define copy_from_kernel_nofault probe_kernel_read
else /* >= 5.8.0 */
  // just do nothing
endif

Reply:
My undergrad Intro to Computer Science professor told me never to define a function in a header file, only declare it and define in a source file. But he was kind of a jerk so what does he know? I suppose I can't come up with a good reason why not to do that, other than my own laziness :). I'll fix it...

Reply:
It should be totally fine to either declare it `static inline` and move it into the header or leave it here and use two `#ifdef` blocks (one for the definition, one for the declaration).

Reply:
I tried to test this on 2.6.25, but my version of make is not compatible with that kernel's makefile apparently. I fell down a rabbit hole of trying to build an older version of make before I realized I didn't care that much. I'll just let the probe builder figure out if it works or not.